admin_guide/node_problem_detector.adoc

```diff
@@ -0,0 +1,319 @@
+
+[NOTE]
+====
+The node problem detector does not detect problems reported by log tools, such as journald or rsyslogd.
+====
+
+[[admin-guide-node-problem-detector-run]]
+== Running the Node Problem Detector
+
+To run the Node Problem Detector first create a xref:../dev_guide/daemonsets#dev-guide-daemonsets[daemonset] object. For {product-title} 
+Google Compute Engine (GCE) infrastructures, see xref:[Running the Node Problem Detector in a GCE Cluster].
+
+When xref:../dev_guide/daemonsets.adoc#dev-guide-daemonsets[creating daemonsets], the `nodeSelector` field indicates the nodes on which the daemonset should deploy.
+
+. Copy the following code to a YAML file on your master:
++
+[source,yaml]
+----
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: node-problem-detector-v0.1
+  namespace: kube-system
+  labels:
+    k8s-app: node-problem-detector
+    version: v0.1
+    kubernetes.io/cluster-service: "true"
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: node-problem-detector
+        version: v0.1
+        kubernetes.io/cluster-service: "true"
+    spec:
+      nodeSelector: <1>
+        type: infra
+      hostNetwork: true
+      containers:
+      - name: node-problem-detector
+        image: k8s.gcr.io/node-problem-detector:v0.1
+        securityContext:
+          privileged: true
+        resources:
+          limits: 
+            cpu: "200m"
+            memory: "100Mi"
+          requests:
+            cpu: "20m"
+            memory: "20Mi"
+        volumeMounts:
+        - name: log 
+          mountPath: /log 
+          readOnly: true
+      volumes:
+      - name: log
+        hostPath: <2>
+          path: /var/log/
+----
++
+<1> The node selector that determines on which nodes pod replicas should be deployed
+<2> Specify the location of your cluster log directory.
+
+. Edit the file for your environment. 
+
+. Save the file as *_node-problem-detector.yaml_*.
+
+. Run the following command to start node problem detector: 
++
+----
+oc create -f <your_yaml_file>.yaml
+----
+
+[[admin-guide-node-problem-detector-addon]]
+== Running the Node Problem Detector as an Add-on Pod
+
+If you have your own cluster bootstrap solution and do not wat to overwrite the default configuration, 
+leverage the add-on pod to further automate the deployment.
+
+Create the xref:admin-guide-node-problem-detector-run[`node-problem-detector.yaml`] file, 
+and save it to the add-on pods directory *_/etc/kubernetes/addons/node-problem-detector_* on master node.
+
+[[admin-guide-node-problem-detector-overwrite]]
+== Overwrite the Configuration
+
+The default configuration is embedded when building the docker image of node problem detector.
+
+You can use a xref:../dev_guide/configmaps.adoc#dev-guide-configmaps[ConfigMap]
+to overwrite the configuration.
+
+. Exit the *_/etc/kubernetes/addons/node-problem-detector/node-problem-detector/config/kernel-monitor.json_* configuraton file.
```

Comment:
@joelsmith Do we use *_/etc/kubernetes/addons/_* directory? If not where does the `kernel-monitor.json` live in OpenShift, if we use it.

Reply:
We don't use the addons framework. For ours, the user needs to edit the configmap provided by the installer:

```
oc edit configmap -n openshift-node-problem-detector node-problem-detector
```

Then edit the section for the `kernel-monitor.json` file. Changes will take effect after running pods are restarted. To restart them, it's easiest to just delete all existing pods:
```
oc delete pods -n openshift-node-problem-detector -l name=node-problem-detector
```